### PR TITLE
Dedicated tab(s) in VS Code panel

### PR DESCRIPTION
**Pull request by xhulz**
Hello Guys!

This is the first version of the log panel (ref #145)

It was a little tricky to develop it because there is not much documentation about it, so I had to dive into some Githubs to understand the way they were using it. But it is something that needs a lot of attention because many times, the events overlap and the synchronization of the client with the back is not so synchronous, causing many of the logs to have to be rendered in a lazy process

Needs a lot of testing and any feedback from you guys would be most welcome :)

For the interface, I built it using Vanilla and HTML. I've read on some forums that using front end framework within the extension can bring some incompatibility issues with the extension itself. Anyway, we can raise another issue (once this one is merged) to implement a front end framework and test it :)

xpto-1660568245906

### CODE REVIEW

File: src/Constants.ts
Comment (acuarica) [inline src/Constants.ts:159]: - Would it be possible to move this into `LogView`, since it's only used there?
- Moreover, could we make it an `enum`?  since it's used that way.

File: src/Constants.ts
Comment (acuarica) [inline src/Constants.ts:155]: Should we convert this into `export const VIEW_TYPE = ...` or something similar to make it simpler?

File: src/views/LogView.ts
Comment (acuarica) [inline src/views/LogView.ts:197]: Not sure if/when this can be an issue, but the `Disposable` returned by `registerCommand` is not included in `context.subscriptions`. See for example `extension.ts::.activate`

**Review by acuarica — APPROVED** [review]
lgtm!

Thanks @xhulz for the hard work. This is a nice feature to have!

**Comment by acuarica** [comment]
Maybe this can be addressed here, but it seems there is a log entry with just a number, looks like a pid. We can either remove the entry or add some additional information.

image

**Comment by xhulz** [comment]
Hey @acuarica!

Thank you for you review :)

Yeah, you right, I'll do these tweaks

Thank you again!

**Comment by xhulz** [comment]
Hey @acuarica,

All your comments was solved :) Thank you again for your eagle eyes! 🦅 

In the case of PID, i was looking at the code and i will raise another issue for that, because we will need to do some modifications on the `spawnProcess` implementation.

Thank you!

**Comment by acuarica** [comment]
Thanks for addressing this @xhulz! Looking good.

File: test/required.test.ts
Comment (michaeljohnbennett) [inline test/required.test.ts:423]: this test is either wrong or the assertion statement is incorrect. Can you fix it.

File: test/required.test.ts
Comment (michaeljohnbennett) [inline test/required.test.ts:467]: this test is either wrong or the assertion statement is incorrect. Can you fix it.

**Review by michaeljohnbennett — CHANGES_REQUESTED** [review]

File: test/required.test.ts
Comment (xhulz) [reply test/required.test.ts]: Hey @michaeljohnbennett, thanks for that :)

Now, we are executing commands inside the Output class `(commands.executeCommand)` so i forgot to change the assertion text to: `"executeVSCommand should be called"`

Now, it's fixed. Thank you again!

**Review by michaeljohnbennett — APPROVED** [review]
perfect, thanks for fixing the tests.

if this is all ok to be merged and put onto the release then I'll let you merge it.

**Comment by acuarica** [comment]
> perfect, thanks for fixing the tests.
> 
> if this is all ok to be merged and put onto the release then I'll let you merge it.

I'd say given that the pre-release is already out there and the changelog has been already updated, to merge this after the final release.

**Comment by xhulz** [comment]
Yeah, it's better :)

**Pull request merged**
